### PR TITLE
feat(loadblance) support to expose service via load balancer

### LINKED ISSUE #178: Support to expose service via load balancer
Currently the only option is ingress. Load balancer is useful for deploying on cloud

### PR DESCRIPTION
**Pull request by Ghostbaby**
support to expose service via load balancer

fix #178

Signed-off-by: zhuhuijun <zhuhuijunzhj@gmail.com>

### CODE REVIEW

File: controllers/goharbor/harbor/loadblance.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+func (r *Reconciler) AddCoreLB(ctx context.Context, harbor *goharborv1alpha2.Harbor, core Core, portal Portal) (CoreLB, error) {
+
+	lbRes, err := r.Controller.AddServiceToManage(ctx, r.GetCoreLB(ctx, harbor), core, portal)
+
+	return CoreLB(lbRes), errors.Wrap(err, "cannot add core lb service")
+}
+
+func (r *Reconciler) GetCoreLB(ctx context.Context, harbor *goharborv1alpha2.Harbor) *corev1.Service {
+	if harbor.Spec.Expose.Core.LoadBalancer == nil {
```
Comment (heww) [inline controllers/goharbor/harbor/loadblance.go:27]: harbor.Spec.Expose.Core.LoadBalancer != nil but harbor.Spec.Expose.Core.LoadBalancer.Enable may be false.

File: controllers/goharbor/harbor/loadblance.go [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+				r.Label("namespace"): namespace,
+			},
+			Type: corev1.ServiceTypeLoadBalancer,
+		},
+	}
+}
+
+type PortalLB graph.Resource
+
+func (r *Reconciler) AddPortalLB(ctx context.Context, harbor *goharborv1alpha2.Harbor, portal Portal) (NotarySignerLB, error) {
```
Comment (steven-zou) [inline controllers/goharbor/harbor/loadblance.go:153]: @Ghostbaby 

It seems functions `addLB` and `getLB` can be general functions that can be used among core,total, notary and singer.

**Review by steven-zou — CHANGES_REQUESTED** [review]
we can do some refactor to the code

File: controllers/goharbor/harbor/loadblance.go (hunk 2 above)
Comment (steven-zou) [reply controllers/goharbor/harbor/loadblance.go:153]: If it is hard to do, then keeping it as it was is ok for me.

**Comment by steven-zou** [comment]
@Ghostbaby per our discussion, let's remove the LB support out of the 1.0 release. You can close this PR.

**Comment by steven-zou** [comment]
Let's find a better way in future to support LB. Close this now.

**Pull request closed without merging**